tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/inheritance_test.dart

```diff
@@ -0,0 +1,249 @@
+import 'package:serverpod_cli/src/analyzer/models/definitions.dart';
+import 'package:serverpod_cli/src/analyzer/models/stateful_analyzer.dart';
+import 'package:serverpod_cli/src/config/experimental_feature.dart';
+import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/model_source_builder.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var config = GeneratorConfigBuilder().withEnabledExperimentalFeatures(
+    [ExperimentalFeature.inheritance],
+  ).build();
+
+  group('Extends property tests', () {
+    test(
+        'Given a child-class of an existing class, then no errors are collected and childClasses and extendsClass are resolved',
+        () {
+      var modelSources = [
+        ModelSourceBuilder().withYaml(
+          '''
+          class: Example
+          fields:
+            name: String
+          ''',
+        ).build(),
+        ModelSourceBuilder().withFileName('example2').withYaml(
+          '''
+          class: ExampleChildClass
+          extends: Example
+          fields:
+            age: int
+          ''',
+        ).build(),
+      ];
+
+      var collector = CodeGenerationCollector();
+      var models =
+          StatefulAnalyzer(config, modelSources, onErrorsCollector(collector))
+              .validateAll();
+
+      expect(
+        collector.errors,
+        isEmpty,
+      );
+
+      var parent = models.first as ClassDefinition;
+      var childClasses = parent.childClasses;
+      var isChildResolved = childClasses.first is ResolvedInheritanceDefinition;
+
+      var child = models.last as ClassDefinition;
+      var extendsClass = child.extendsClass;
+      var isExtendsResolved = extendsClass is ResolvedInheritanceDefinition;
+
+      expect(isChildResolved, isTrue);
+      expect(isExtendsResolved, isTrue);
```

Comment:
Suggestion: Make this into a group and two destinct tests, where each field is in it's own test. 

While related I think these are two different values / features that could fail independently and it will be easier to debug / fix if they are reported as destinct tests.